src/pages/workspaces/workspace/analysis/AnalysisLauncher.js

```diff
@@ -88,7 +90,7 @@ const AnalysisLauncher = _.flow(
         mode && h(RuntimeKicker, { runtime: currentRuntime, refreshRuntimes }),
         mode && h(RuntimeStatusMonitor, { runtime: currentRuntime }),
         h(ComputeModal, {
-          isOpen: createOpen,
+          isOpen: createOpen && !workspace?.azureContext,
```

Comment:
Nit: A variable along the lines of `isAzureWorkspace = !!workspace?.azureContext` might make this read a little clearer.

Reply:
Note though that if the workspace is undefined, the value for this boolean will not be correct. We can only determine if something is an Azure vs. Google workspace once we have loaded the workspace details and workspace is not undefined.

If it's legitimate that workspace could be undefined, you need to handle this possibility. Or if you are sure that workspace will be defined, you should remove the `?` because the usage implies that it is possible for workspace to be undefined.

Reply:
Ahh I was misusing the "?" operator. I thought I'd get a key error by trying to access azureContext from workspace if it didn't exist.

New implementation incorporates both feedback.

Reply:
Yes, I'm concerned that people just toss "?" on variables all the time because there is this perception that it makes everything safe. But it only works if your code is written in such a way that it can handle the result being undefined.